cns/restserver/ipam.go

```diff
@@ -92,40 +92,54 @@ func (service *HTTPRestService) releaseIPConfigHandler(w http.ResponseWriter, r
 	return
 }
 
-// MarkIPAsPendingRelease will mark IPs to pending release state.
-func (service *HTTPRestService) MarkIPAsPendingRelease(numberToMark int) (map[string]cns.IPConfigurationStatus, error) {
-	allReleasedIPs := make(map[string]cns.IPConfigurationStatus)
-	// Ensure PendingProgramming IPs will be release before Available ones.
-	ipStateTypes := [2]string{cns.PendingProgramming, cns.Available}
-
+// MarkIPAsPendingRelease will set the IPs which are in PendingProgramming or Available to PendingRelease state
+// It will try to update [totalIpsToRelease]  number of ips.
+func (service *HTTPRestService) MarkIPAsPendingRelease(totalIpsToRelease int) (map[string]cns.IPConfigurationStatus, error) {
+	pendingReleasedIps := make(map[string]cns.IPConfigurationStatus)
 	service.Lock()
 	defer service.Unlock()
-	for _, ipStateType := range ipStateTypes {
-		pendingReleaseIPs := service.markSpecificIPTypeAsPending(numberToMark, ipStateType)
-		for uuid, pependingReleaseIP := range pendingReleaseIPs {
-			allReleasedIPs[uuid] = pependingReleaseIP
-		}
-		numberToMark -= len(pendingReleaseIPs)
-		if numberToMark == 0 {
-			return allReleasedIPs, nil
-		}
-	}
-	return nil, fmt.Errorf("Failed to mark %d IP's as pending, only marked %d IP's", numberToMark, len(allReleasedIPs))
-}
 
-func (service *HTTPRestService) markSpecificIPTypeAsPending(numberToMark int, ipStateType string) map[string]cns.IPConfigurationStatus {
-	pendingReleaseIPs := make(map[string]cns.IPConfigurationStatus)
-	for uuid, mutableIPConfig := range service.PodIPConfigState {
-		if mutableIPConfig.State == ipStateType {
-			mutableIPConfig.State = cns.PendingRelease
-			service.PodIPConfigState[uuid] = mutableIPConfig
-			pendingReleaseIPs[uuid] = mutableIPConfig
-			if len(pendingReleaseIPs) == numberToMark {
-				return pendingReleaseIPs
+	for uuid, ipcConfig := range service.PodIPConfigState {
+        if ipcConfig.State == cns.PendingProgramming {
+        	if err := service.updateIPConfigState(uuid, cns.PendingRelease); err != nil {
+                return nil, err
+            }
+
+            pendingReleasedIps[uuid] = ipcConfig
+			if len(pendingReleasedIps) == totalIpsToRelease {
+				return pendingReleasedIps, nil
```

Comment:
there's still something off with formatting in this file. Do you have gofmt or goimports enabled on save?

Reply:
it is enabled, not sure why it is off (at least not showing on my CodeFlow or GoLand

---
In reply to: [585063214](https://github.com/Azure/azure-container-networking/pull/806#discussion_r585063214) [](ancestors = 585063214)